src/mscorlib/src/System/String.cs

```diff
@@ -666,6 +666,37 @@ private unsafe String CtorCharPtrStartLength(char* ptr, int startIndex, int leng
         [MethodImplAttribute(MethodImplOptions.InternalCall)]
         public extern String(char c, int count);
 
+        private unsafe string CtorCharRefLength(ref char value, int length)
+        {
+            Debug.Assert(length >= 0);
+
+            if (length == 0)
+            {
+                return Empty;
+            }
+
+            try
+            {
+                string result = FastAllocateString(length);
+                fixed (char* dest = &result._firstChar, src = &value)
+                {
+                    wstrcpy(dest, src, length);
+                }
+                return result;
+            }
+            catch (NullReferenceException)
```

Comment:
I do not think this catch for `NullReferenceException` should be here.

If you create a Span from unmanaged memory, you have to guarantee that the memory is valid. We should not be protecting in BCL against cases where you get it wrong.